File: src/main/java/rk/cinema/model/ReservedSeat.java
package rk.cinema.model;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.Relation;
import rk.cinema.enums.TicketType;

@MappedEntity
public record ReservedSeat(
        @Id
        @GeneratedValue
        Long id,

        @Relation(Relation.Kind.MANY_TO_ONE)
        @NonNull
        Seat seat,

        @Relation(Relation.Kind.MANY_TO_ONE)
        @NonNull
        Reservation reservation,

        @NonNull
        TicketType ticketType
) {
}

File: src/main/kotlin/rk/cinema/model/ReservedSeat.kt
package rk.cinema.model

import io.micronaut.core.annotation.NonNull
import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.Relation
import rk.cinema.enums.TicketType

@MappedEntity("reserved_seat")
data class ReservedSeat(
    @GeneratedValue @param:Id val id: Long,
    @Relation(Relation.Kind.MANY_TO_ONE) val seat: @NonNull Seat,
    @Relation(Relation.Kind.MANY_TO_ONE) val reservation: @NonNull Reservation,
    val ticketType: @NonNull TicketType,
)
